helix-view/src/clipboard.rs

```diff
@@ -243,12 +242,37 @@ pub mod provider {
 
         #[cfg(not(any(windows, target_os = "macos")))]
         pub fn is_exit_success(program: &str, args: &[&str]) -> bool {
-            std::process::Command::new(program)
-                .args(args)
-                .output()
-                .ok()
-                .and_then(|out| out.status.success().then(|| ())) // TODO: use then_some when stabilized
-                .is_some()
+            use helix_lsp::block_on;
+            use tokio::process::Command;
+            use tokio::time::timeout;
+
+            async fn is_exit_success_async(program: &str, args: &[&str]) -> bool {
```

Comment:
Hmm, using async might actually be problematic since tests might call this function outside of a tokio runtime.

Reply:
Oh no :( well, I made the changes for the rest, but let me know if you want this changed to be sync.

Reply:
Maybe it would be better to expose the method as an async fn then let the caller use block_on as needed? I'm just worried about the method looking sync but internally blocking on async code

Reply:
Yeah, that makes sense. I think what was messing with me was the cfgs, I didn't want a linux version to be async while others sync, but I see now that this function is linux only. I pushed new changes, is this what you had in mind?

Reply:
Also, [tokio::process::Command](https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.output) mentions that it doesn't have the same exact behavior as std::process::Command concerning stdout/stderr pipes, is this something we need to worry about?